diff --git a/DnsServerCore/Dns/Applications/DnsApplicationAssemblyLoadContext.cs b/DnsServerCore/Dns/Applications/DnsApplicationAssemblyLoadContext.cs
index 0ea273e7..cf2be8df 100644
--- a/DnsServerCore/Dns/Applications/DnsApplicationAssemblyLoadContext.cs
+++ b/DnsServerCore/Dns/Applications/DnsApplicationAssemblyLoadContext.cs
@@ -69,16 +69,9 @@ namespace DnsServerCore.Dns.Applications
 
                 foreach (Assembly loadedAssembly in Default.Assemblies)
                 {
-                    if (!string.IsNullOrEmpty(loadedAssembly.Location))
-                    {
-                        if (requiredAssemblyName.Equals(Path.GetFileNameWithoutExtension(loadedAssembly.Location), StringComparison.OrdinalIgnoreCase))
-                            return loadedAssembly;
-                    }
-                    else
-                    {
-                        if (requiredAssemblyName.Equals(loadedAssembly.GetName().Name, StringComparison.OrdinalIgnoreCase))
-                            return loadedAssembly;
-                    }
+                    string assemblyLocation = !string.IsNullOrEmpty(loadedAssembly.Location) ? loadedAssembly.Location : Path.Combine(AppContext.BaseDirectory, loadedAssembly.GetName().Name + ".dll");
+                    if (requiredAssemblyName.Equals(Path.GetFileNameWithoutExtension(assemblyLocation), StringComparison.OrdinalIgnoreCase))
+                        return loadedAssembly;
                 }
 
                 return null;
@@ -96,21 +89,11 @@ namespace DnsServerCore.Dns.Applications
 
                 foreach (Assembly loadedAssembly in loadedAssemblies)
                 {
-                    if (!string.IsNullOrEmpty(loadedAssembly.Location))
+                    string assemblyLocation = !string.IsNullOrEmpty(loadedAssembly.Location) ? loadedAssembly.Location : Path.Combine(AppContext.BaseDirectory, loadedAssembly.GetName().Name + ".dll");
+                    if (dllFileName.Equals(Path.GetFileNameWithoutExtension(assemblyLocation), StringComparison.OrdinalIgnoreCase))
                     {
-                        if (dllFileName.Equals(Path.GetFileNameWithoutExtension(loadedAssembly.Location), StringComparison.OrdinalIgnoreCase))
-                        {
-                            isLoaded = true;
-                            break;
-                        }
-                    }
-                    else
-                    {
-                        if (dllFileName.Equals(loadedAssembly.GetName().Name, StringComparison.OrdinalIgnoreCase))
-                        {
-                            isLoaded = true;
-                            break;
-                        }
+                        isLoaded = true;
+                        break;
                     }
                 }
 
diff --git a/DnsServerCore/DnsWebService.cs b/DnsServerCore/DnsWebService.cs
index 798d19ab..28e930de 100644
--- a/DnsServerCore/DnsWebService.cs
+++ b/DnsServerCore/DnsWebService.cs
@@ -130,7 +130,7 @@ namespace DnsServerCore
             Assembly assembly = Assembly.GetExecutingAssembly();
 
             _currentVersion = assembly.GetName().Version;
-            _appFolder = Path.GetDirectoryName(assembly.Location);
+            _appFolder = AppContext.BaseDirectory;
 
             if (configFolder is null)
                 _configFolder = Path.Combine(_appFolder, "config");
