diff --git a/DnsServerCore/Dns/Applications/DnsApplicationAssemblyLoadContext.cs b/DnsServerCore/Dns/Applications/DnsApplicationAssemblyLoadContext.cs
index 0ea273e7..195d1721 100644
--- a/DnsServerCore/Dns/Applications/DnsApplicationAssemblyLoadContext.cs
+++ b/DnsServerCore/Dns/Applications/DnsApplicationAssemblyLoadContext.cs
@@ -69,16 +69,8 @@ namespace DnsServerCore.Dns.Applications
 
                 foreach (Assembly loadedAssembly in Default.Assemblies)
                 {
-                    if (!string.IsNullOrEmpty(loadedAssembly.Location))
-                    {
-                        if (requiredAssemblyName.Equals(Path.GetFileNameWithoutExtension(loadedAssembly.Location), StringComparison.OrdinalIgnoreCase))
-                            return loadedAssembly;
-                    }
-                    else
-                    {
-                        if (requiredAssemblyName.Equals(loadedAssembly.GetName().Name, StringComparison.OrdinalIgnoreCase))
-                            return loadedAssembly;
-                    }
+                    if (requiredAssemblyName.Equals(loadedAssembly.GetName().Name, StringComparison.OrdinalIgnoreCase))
+                        return loadedAssembly;
                 }
 
                 return null;
@@ -96,21 +88,10 @@ namespace DnsServerCore.Dns.Applications
 
                 foreach (Assembly loadedAssembly in loadedAssemblies)
                 {
-                    if (!string.IsNullOrEmpty(loadedAssembly.Location))
+                    if (dllFileName.Equals(loadedAssembly.GetName().Name, StringComparison.OrdinalIgnoreCase))
                     {
-                        if (dllFileName.Equals(Path.GetFileNameWithoutExtension(loadedAssembly.Location), StringComparison.OrdinalIgnoreCase))
-                        {
-                            isLoaded = true;
-                            break;
-                        }
-                    }
-                    else
-                    {
-                        if (dllFileName.Equals(loadedAssembly.GetName().Name, StringComparison.OrdinalIgnoreCase))
-                        {
-                            isLoaded = true;
-                            break;
-                        }
+                        isLoaded = true;
+                        break;
                     }
                 }
 
diff --git a/DnsServerCore/DnsWebService.cs b/DnsServerCore/DnsWebService.cs
index 798d19ab..c8025851 100644
--- a/DnsServerCore/DnsWebService.cs
+++ b/DnsServerCore/DnsWebService.cs
@@ -130,10 +130,10 @@ namespace DnsServerCore
             Assembly assembly = Assembly.GetExecutingAssembly();
 
             _currentVersion = assembly.GetName().Version;
-            _appFolder = Path.GetDirectoryName(assembly.Location);
+            _appFolder = "/usr/share/technitium";
 
             if (configFolder is null)
-                _configFolder = Path.Combine(_appFolder, "config");
+                _configFolder = "/etc/technitium";
             else
                 _configFolder = configFolder;
 
